geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -1605,21 +1605,34 @@ protected Profile instantiateProfile(InternalDistributedMember memberId, int ver
       InternalCache cache = getProxyBucketRegion().getCache();
       List<CacheServer> servers = cache.getCacheServers();
 
-      HashSet<BucketServerLocation66> serverLocations = new HashSet<>();
-      for (CacheServer cacheServer : servers) {
-        CacheServerImpl server = (CacheServerImpl) cacheServer;
+      HashSet<BucketServerLocation66> serverLocations = getBucketServerLocations(version, servers);
+      if (serverLocations.size() > 0) {
+        return new ServerBucketProfile(memberId, version, getBucket(), serverLocations);
+      }
+    }
+    return new BucketProfile(memberId, version, getBucket());
+  }
+
+  protected HashSet<BucketServerLocation66> getBucketServerLocations(int version,
+      List<CacheServer> servers) {
+    HashSet<BucketServerLocation66> serverLocations = new HashSet<>();
+    for (CacheServer cacheServer : servers) {
+      CacheServerImpl server = (CacheServerImpl) cacheServer;
+      try {
         if (server.isRunning() && (server.getExternalAddress() != null)) {
```

Comment:
call getExternalAddress once instead of twice

Reply:
Done